workflow/cron/operator.go

```diff
@@ -159,19 +169,45 @@ func (woc *cronWfOperationCtx) runOutstandingWorkflows() error {
 	return nil
 }
 
+func (woc *cronWfOperationCtx) reconcileDeletedWfs() error {
+	wfList, err := woc.wfLister.List()
+	if err != nil {
+		return fmt.Errorf("unable to list workflows: %s", err)
+	}
+
+	currentWfs := make(map[types.UID]bool)
+	for _, wf := range wfList {
+		currentWfs[wf.UID] = true
+	}
+
+	for _, objectRef := range woc.cronWf.Status.Active {
+		if found := currentWfs[objectRef.UID]; !found {
+			woc.removeFromActiveList(objectRef.UID)
+		}
+	}
+
+	return nil
+}
+
 func (woc *cronWfOperationCtx) removeActiveWf(wf *v1alpha1.Workflow) {
 	if wf == nil || wf.ObjectMeta.UID == "" {
 		return
 	}
-	for i, objectRef := range woc.cronWf.Status.Active {
-		if objectRef.UID == wf.ObjectMeta.UID {
-			woc.cronWf.Status.Active = append(woc.cronWf.Status.Active[:i], woc.cronWf.Status.Active[i+1:]...)
-			err := woc.persistUpdate()
-			if err != nil {
-				log.Errorf("Unable to update CronWorkflow '%s': %s", woc.cronWf.Name, err)
-			}
+	woc.removeFromActiveList(wf.ObjectMeta.UID)
+	err := woc.persistUpdate()
+	if err != nil {
+		log.Errorf("Unable to update CronWorkflow '%s': %s", woc.cronWf.Name, err)
```

Comment:
minor - should be structured logging

Reply:
Perhaps do this in a separate PR? A significant amount of logging statements would have to change, and I think it would be better to do them all at once